### ISSUE TITLE
drivers: video: ov2640: migrate to Frame Interval API

### ISSUE THREAD

**Issue by josuah** [zephyrproject-rtos/zephyr#80927]
**Is your enhancement proposal related to a problem? Please describe.**

https://github.com/zephyrproject-rtos/zephyr/pull/71962 introduced clock rate control, which controls the frame rate, but it uses the devicetree rather than the video API dedicated to it.

**Describe the solution you'd like**

Migrate the ov2640 to the Frame Interval-based API introduced in https://github.com/zephyrproject-rtos/zephyr/pull/72254 and remove the devicetree config.

**Describe alternatives you've considered**

Keep the devicetree for providing a default, and also introduce the API, but that is less compelling.

**Additional context**

https://github.com/zephyrproject-rtos/zephyr/pull/72254 got introduced after ov2640: there was no API for controlling frame rate at the time.

**Comment by ngphibang** [comment]
Good idea. The clock_rate_control in ov2640 is the original thing that leads to the need of the frame interval API. @CharlesDias
